HW03-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,34 @@
+package homework;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.AbstractMap;
+
+public class CustomerService {
+
+    private static Comparator<Customer> byScores = (c1, c2) -> Long.compare(c1.getScores(), c2.getScores());
+    private static TreeMap <Customer, String> customers = new TreeMap<>(byScores);
```

Comment:
Слева тип лучше объявлять как интерфейс:
```suggestion
    private static NavigableMap<Customer, String> customers = new TreeMap<>(byScores);
```